evaluator/evaluate.go

```diff
@@ -452,11 +453,14 @@ func run(readerBuilder mqueue.CreateReader) {
 	// We create multiple consumers, and hope that the partition rebalancing
 	// algorithm assigns each consumer a single partition
 	for i := 0; i < consumerCount; i++ {
-		go mqueue.RunReader(evalTopic, readerBuilder, handler)
+		wg.Add(1)
```

Comment:
Why increment here ? The RunReader should manage its contribution to the waitgroup.

Reply:
If we put it there (inside go routine function) theoretically wg.Wait() calling in main thread could be reached earlier than first wg.Add(1) is called.